### ISSUE TITLE
Search functionality

### ISSUE THREAD

**Issue by m-kuhn** [opengisch/QField#165]
The search should be based on a layers "display expression". The search is case insensitive and match substrings.

The search results are limited to 20 entries at a time.

When clicking on a search result, the map will be located at the position of the result.
A button will be placed next to the result which allows to open the feature form instead of locating the map at the result.

**Comment by saemiw** [comment]
Sound interesting, why is the search limited to 20 entries? You mean twenty entries that are returned that match the search criteria and not 20 searching criteria.

**Comment by 3nids** [comment]
yes, it's the number of results which is limited to avoid loading thousands of entries.
There is no clear decision on the UI yet, but a right panel (similar to the feature form) sounds like a good idea.

**Comment by BeaSavoie** [comment]
Hi !
How can we search in QField ? Is the present description implemented in the current version of QField ?  
What are the problems now to realize a search window to the right as features ? 
We are very interested in a search functionnality and ready to help its development.
Thanks for your return...

**Comment by 3nids** [comment]
The search functionality is not released yet but should be in a near future.
It will be a simple single text based search (like in google maps for instance).
There is no plan yet to implement a feature-form based search. Is it what you are looking for?

**Comment by BeaSavoie** [comment]
Yes it's exactly what we need. 
Users of our Qfield projects would like to search a name, a ID which exist on attributes. We have find one or two alternatives but it's not like a search form...
Courchevel's services can participate to the funding plan. I think we are not alone to want those functionality and ready to pay a part of its implementation.
Thanks...
